include/util/conditional_restrictions.hpp

```diff
@@ -0,0 +1,118 @@
+#ifndef CONDITIONAL_RESTRICTIONS_HPP
+#define CONDITIONAL_RESTRICTIONS_HPP
+
+//#define BOOST_SPIRIT_DEBUG
+
+#include <boost/fusion/include/adapt_struct.hpp>
+#include <boost/spirit/include/phoenix.hpp>
+#include <boost/spirit/include/qi.hpp>
+
+namespace osrm
+{
+namespace util
+{
+
+// Helper functions for OSM conditional restrictions
+// http://wiki.openstreetmap.org/wiki/Conditional_restrictions
+// Consitional restrictions is a vector of ConditionalRestriction
+// with a restriction value and a condition string
+struct ConditionalRestriction
+{
+    std::string value;
+    std::string condition;
+};
+
+// TODO: remove debug
+// inline std::ostream &operator<<(std::ostream &stream, const ConditionalRestriction &restriction)
+// {
+//     return stream << restriction.value << "=" << restriction.condition;
+// }
```

Comment:
Feel free to keep it - but I would uncomment it.

Also seeing that you adapt the struct, fusion should give you this I/O operator for free

http://www.boost.org/doc/libs/1_63_0/libs/fusion/doc/html/fusion/sequence/operator.html